proto/trip.proto

```diff
@@ -119,6 +119,25 @@ message TripLeg {
     uint32 type = 1;
   }
 
+  message ConditionalSpeedLimit {
+    uint32 speed_limit = 1;
+
+    // Encoded condition with the following format:
+    // uint64_t day_type_ : 1;       // type of day_dow, 0 - day of month, 1 - nth day of week
+    // uint64_t dow_ : 7;           // day of week mask, e.g. 0x0111110 for Mo-Fr as week starts from Su
+    // uint64_t begin_hrs_ : 5;     // begin hours, 0 if not set
+    // uint64_t begin_mins_ : 6;    // begin minutes, 0 if not set
+    // uint64_t begin_month_ : 4;   // begin month, from 1 to 12, 0 if not set
+    // uint64_t begin_day_dow_ : 5; // begin day of month or nth dow, i.e. 1st Sunday
+    // uint64_t begin_week_ : 3;    // which week does this start, i.e. 1st week in Oct
+    // uint64_t end_hrs_ : 5;       // end hours, 0 if not set
+    // uint64_t end_mins_ : 6;      // end minutes, 0 if not set
+    // uint64_t end_month_ : 4;     // end month, from 1 to 12, 0 if not set
+    // uint64_t end_day_dow_ : 5;   // end day of month or nth dow, i.e. last Sunday
+    // uint64_t end_week_ : 3;      // which week does this end, i.e. last week in Oct
+    uint64 condition = 2;
```

Comment:
The string format is really strange and really hard to parse, so downstream users will benefit from the bitfield format that then can be used on devices to determine which speed limit to show.
But I'm open for ideas how to do it better.

Reply:
Okay, surprisingly, it is extremely hard to use it in a languages that do not support bitfields (like Python).

Reply:
yeah i had originally suggested a bitfield for this (or one of the other restriction formats) to @gknisely but he was like "this is a massive pain im not doing it" hahaha. and yeah for python you'd have to do the whole `ctype` struct to get a bitfield to work. it is quite verbose without the bitfield (ie a json field for each integer) but then again the locate output only comes with all this information when you specify `verbose=true` so maybe its ok?

Reply:
oh and if you are talking about the bitfield stored in a single integer in proto and then consuming the proto format as output from the service then i think sadly its better to force the requester to figure out the bitfield than to bloat the proto with all the various fields

Reply:
Well, I've played a bit with bits unpacking like this
```python
import ctypes
class TimeDomain(ctypes.Structure):
    _fields_ = [
        ("day_dow_type", ctypes.c_uint64, 1),
        ("dow", ctypes.c_uint64, 7),
        ("begin_hrs", ctypes.c_uint64, 5),
        ("begin_mins", ctypes.c_uint64, 6),
        ("begin_month", ctypes.c_uint64, 4),
        ("begin_day_dow", ctypes.c_uint64, 5),
        ("begin_week", ctypes.c_uint64, 3),
        ("end_hrs", ctypes.c_uint64, 5),
        ("end_mins", ctypes.c_uint64, 6),
        ("end_month", ctypes.c_uint64, 4),
        ("end_day_dow", ctypes.c_uint64, 5),
        ("end_week", ctypes.c_uint64, 3),
        ("spare", ctypes.c_uint64, 10),
    ]
td = TimeDomain.from_buffer_copy(ctypes.c_uint64(12884906752))
```

and the biggest problem with it is that you need additional tests to cover how value is unpacked. 
In the de685b9601afdffe0b9b9b61507a74f98dea3c0e I've added a `message TimeDomain` and it doesn't look that bad, while the downstream usage seems much more convenient.

Reply:
> yeah i had originally suggested a bitfield for this (or one of the other restriction formats) to @gknisely but he was like "this is a massive pain im not doing it" hahaha. and yeah for python you'd have to do the whole `ctype` struct to get a bitfield to work. it is quite verbose without the bitfield (ie a json field for each integer) but then again the locate output only comes with all this information when you specify `verbose=true` so maybe its ok?

Not a pain...just no time

Reply:
sorry yeah thats all i meant